### PR TITLE
Pytorch 1.0 patch

### PR DESCRIPTION
**Pull request by Tomcli**
- Added pytorch 0.4.1 and 1.0 distributed examples at etc/examples.
- The 2 main distributed learning example for Pytorch 1.0 is [c10d-Parallelism-example](https://github.com/IBM/FfDL/tree/pytorch-patch/etc/examples/c10d-Parallelism-example) and [c10d-onnx-mpi](https://github.com/IBM/FfDL/tree/pytorch-patch/etc/examples/c10d-onnx-mpi)
- Added Seldon's ONNX serving example using nGraph in the Seldon community folder.
- LCM code changes to make pytorch MPI backend compliance. 
 
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

File: etc/examples/c10d-Parallelism-example/README.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
+# PyTorch Fashion MNIST distributed training example using native data parallelism
+
+This is a 2 convolutional layers model using native pytorch method **dist.init_process_group** and **DistributedDataParallel** to sync and join all the worker nodes.
+- Tested with Syncing using **Shared File System** where worker processes will share their location/group name information on a shared file. (currently using it on shared pvc among the ffdl learners)
+- Tested with GPU with both **Gloo** and **NCCL** Backend.
+- This example is able to handle multiple GPUs in one rank.
```
Comment (sdmonov) [inline etc/examples/c10d-Parallelism-example/README.md]: I think we should mention also that this example will assign single rank per node only. This is mentioned into the top level list of examples but will be nice to have it here too.

File: etc/examples/c10d-Parallelism-example/README.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+```shell
+restapi_port=$(kubectl get service ffdl-restapi -o jsonpath='{.spec.ports[0].nodePort}')
+export DLAAS_URL=http://$PUBLIC_IP:$restapi_port; export DLAAS_USERNAME=test-user; export DLAAS_PASSWORD=test;
+
+# Obtain the correct CLI for your machine
+CLI_CMD=$(pwd)/cli/bin/ffdl-$(if [ "$(uname)" = "Darwin" ]; then echo 'osx'; else echo 'linux'; fi)
+```
+
+## Step 2 - Creating a Manifest File
+Create the code that will train your model. We will use `train_dist_onnx.py` in the folder "pytorch-dist-onnx" as an example
```
Comment (sdmonov) [inline etc/examples/c10d-Parallelism-example/README.md]: I think we should change the name of the training python script and the folder to reflect the examples: `train_dist_parallel.py` and "c10d-Parallelism-example" for example

File: etc/examples/c10d-Parallelism-example/README.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+Create a .yml file with the necessary information. manifest.yml has further instructions on creating an appropriate manifest file.
+
+## Step 3a - Deploying the training job to FfDL
+
+We will now go back to this directory and deploy our training job to FfDL using the path to the .yml and path to the folder containing the experiment.py
+```bash
+cd <path to this demo repo>/pytorch-dist-onnx
```
Comment (sdmonov) [inline etc/examples/c10d-Parallelism-example/README.md]: change the folder name to reflect the folder name used in Step 2

File: etc/examples/c10d-Parallelism-example/README.md [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+# Replace manifest.yml with the path to your .yml file
+# Replace Image with the path to the folder containing your file created in step 6
```
Comment (sdmonov) [inline etc/examples/c10d-Parallelism-example/README.md]: there is no step 6 in this example. we should remove this line

File: etc/examples/c10d-Parallelism-example/README.md (hunk 1 above)
Comment (animeshsingh) [reply etc/examples/c10d-Parallelism-example/README.md]: @sdmonov Makes sense.

File: etc/examples/c10d-Parallelism-example/README.md (hunk 3 above)
Comment (animeshsingh) [reply etc/examples/c10d-Parallelism-example/README.md]: @sdmonov @Tomcli also would recommend making folders lowercase

File: etc/examples/c10d-Parallelism-example/README.md [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+$CLI_CMD train manifest.yml model-files
+```
+## Step 3b - Deploying the training job to FfDL using the FfDL UI
+
+Alternatively, the FfDL UI can be used to deploy jobs. First zip your model.
+```bash
+# Replace fashion-training with the path to your training file folder
+# Replace fashion-training.zip with the path where you want the .zip file stored
```
Comment (sdmonov) [inline etc/examples/c10d-Parallelism-example/README.md]: change fashion-training to fashion-data-parallel to be consistent with the line 87.

File: etc/examples/c10d-Parallelism-mpi/README.md [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+```shell
+restapi_port=$(kubectl get service ffdl-restapi -o jsonpath='{.spec.ports[0].nodePort}')
+export DLAAS_URL=http://$PUBLIC_IP:$restapi_port; export DLAAS_USERNAME=test-user; export DLAAS_PASSWORD=test;
+
+# Obtain the correct CLI for your machine
+CLI_CMD=$(pwd)/cli/bin/ffdl-$(if [ "$(uname)" = "Darwin" ]; then echo 'osx'; else echo 'linux'; fi)
+```
+
+## Step 2 - Creating a Manifest File
+Create the code that will train your model. We will use `train_dist_onnx.py` in the folder "pytorch-dist-onnx" as an example
```
Comment (sdmonov) [inline etc/examples/c10d-Parallelism-mpi/README.md]: fix the naming of the the script and the folder to reflect the example

File: etc/examples/c10d-Parallelism-mpi/README.md [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+
+Create a .yml file with the necessary information. manifest.yml has further instructions on creating an appropriate manifest file.
+
+## Step 3a - Deploying the training job to FfDL
+
+We will now go back to this directory and deploy our training job to FfDL using the path to the .yml and path to the folder containing the experiment.py
+```bash
+cd <path to this demo repo>/pytorch-dist-onnx
```
Comment (sdmonov) [inline etc/examples/c10d-Parallelism-mpi/README.md]: fix the name of the folder to reflect the example

**Comment by sdmonov** [comment]
Few more things:

* There is this `Replace Image with the path to the folder containing your file created in step 6` in few of the examples that needs to be removed from there (just copy/paste issue I'm sure)
* check the examples READMEs for the names of the scripts and the folders
* rename c10d-Parallelism-mpi to lower case (per @animeshsingh request from earlier)

**Comment by Tomcli** [comment]
Thanks @sdmonov for the review, I updated the readme with the correct naming and make it consistent with the current branch.

**Comment by animeshsingh** [comment]
LGTM

**Pull request merged**